### ISSUE TITLE
dra_manager_state checkpoint error

### ISSUE THREAD

**Issue by lengrongfu** [kubernetes/kubernetes#127653]
### What happened?

1. start i create a cluster use kubernetes v1.30.0, cluster running success.
2. and then, i upgrade cluster version to v1.31.1, but kubelet can't start. 

error info is: `failed to get checkpoint dra_manager_state: checkpoint is corrupted`

![image](https://github.com/user-attachments/assets/6e6a1e79-7531-429d-bd49-886645647ee3)

### What did you expect to happen?

Hope it start success use kubelet v1.31.1 version.

### How can we reproduce it (as minimally and precisely as possible)?

1. deploy a 1.30.0 cluster
2. upgrader to v.131.1

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

root@ubuntu:/etc/kubeasz/bin# kubectl version
Client Version: v1.31.1
Kustomize Version: v5.4.2
Server Version: v1.31.1

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by lengrongfu** [comment]
@bart0sh Can you take a look at this? Is this what we expected? thanks~

**Comment by lengrongfu** [comment]
thanks you answer.

Yes, removing the previous file can solve this problem, what I am discussing is whether this is a system bug? If so, how should we fix it.

**Comment by bart0sh** [comment]
@lengrongfu 
> Can you take a look at this? Is this what we expected?

It looks like a duplicate of https://github.com/kubernetes/kubernetes/issues/123552. It's fixed in 1.32.
